internal/driverutil/description_test.go

```diff
@@ -0,0 +1,1282 @@
+package driverutil
+
+import (
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"path"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/internal/assert"
+	"go.mongodb.org/mongo-driver/internal/require"
+	"go.mongodb.org/mongo-driver/internal/spectest"
+	"go.mongodb.org/mongo-driver/mongo/address"
+	"go.mongodb.org/mongo-driver/mongo/readpref"
+	"go.mongodb.org/mongo-driver/tag"
+	"go.mongodb.org/mongo-driver/x/mongo/driver/description"
+)
+
+// TODO(GODRIVER-33): Use proper extended JSON parsing to eliminate the need for this struct.
+type lastWriteDateInner struct {
+	Value string `json:"$numberLong"`
+}
```

Comment:
Optional: We can remove this if we use the Extended JSON unmarshaler instead of the built-in JSON unmarshaler as recommended below.